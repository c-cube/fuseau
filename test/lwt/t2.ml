let pf = Printf.printf
let ( let@ ) = ( @@ )

let run () =
  Fuseau_lwt.main @@ fun () ->
  pf "start\n%!";
  let f1 =
    Fuseau.spawn ~name:"f1" (fun () ->
        Fuseau.sleep_s 0.5;
        pf "f1 returns\n%!";
        41)
  in
  let f2 =
    Fuseau.spawn ~name:"f2" (fun () ->
        Fuseau.sleep_s 0.3;
        pf "f2 returns\n%!";
        100)
  in

  Trace.message "await both";
  pf "await both\n%!";
  Fuseau.await f1 + Fuseau.await f2

let () =
  let@ () = Trace_tef.with_setup () in
  let n = run () in
  pf "f2 returns\n%!";
  pf "res: %d\n" n
