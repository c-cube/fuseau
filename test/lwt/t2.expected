start
await both
f2 returns
f1 returns
f2 returns
res: 141
